brozzler/ydl.py

```diff
@@ -418,6 +420,43 @@ def _try_youtube_dl(worker, ydl, site, page):
     return ie_result
 
 
+def get_video_captures(site, source="youtube"):
+    if not VIDEO_DATA_SOURCE:
+        return None
+
+    if VIDEO_DATA_SOURCE and VIDEO_DATA_SOURCE.startswith("postgresql"):
+        import psycopg
+
+        account_id = site.account_id if site.account_id else None
+        seed = site.metadata.ait_seed_id if site.metadata.ait_seed_id else None
+        if source == "youtube":
+            containing_page_url_pattern = "http://youtube.com/watch"  # yes, video data canonicalization uses "http"
+        # support other sources here
+        else:
+            containing_page_url_pattern = None
+        if account_id and seed and source:
+            pg_query = (
+                "SELECT distinct(containing_page_url) from video where account_id = %s and seed = %s and containing_page_url like %s",
+                (
+                    account_id,
+                    seed,
+                    containing_page_url_pattern,
+                ),
+            )
+        elif seed and source:
+            pg_query = (
+                "SELECT containing_page_url from video where seed = %s and containing_page_url like %s",
+                (seed, containing_page_url_pattern),
+            )
+        else:
+            return None
+        with psycopg.connect(VIDEO_DATA_SOURCE) as conn:
+            with conn.cursor(row_factory=psycopg.rows.scalar_row) as cur:
+                cur.execute(pg_query)
+                return cur.fetchall()
+    return None
```

Comment:
We should consider wrapping this call with an interface as a client
- This would allow us to have the client use pg directly or make an API call
- A client could maintain connections instead re-connecting to PG on every check
- We need to add unit tests before we ship this feature, a client might make unit testing easier

Reply:
recent updates include most of this work